include/libtorrent/alert_types.hpp

```diff
@@ -56,6 +56,9 @@ POSSIBILITY OF SUCH DAMAGE.
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
 
+// c++/cx keyword
+#undef interface
```

Comment:
how about checking __cplusplus_winrt btw? It doesn't solve the problem, but it limits the impact. Do you think using #pragma unmanaged could be an alternative to undefining "interface"? https://msdn.microsoft.com/en-us/library/0adb9zxe.aspx